app/controllers/user/listings_controller.rb

```diff
@@ -20,8 +20,8 @@ def create
       params["listing_images"]["images"].each do |i|
         @listing.listing_images.build(image: i)
       end
-    end
-    if @listing.save
+    elsif @listing.save
+      current_user.update_attributes(role: 1)
```

Comment:
The way this is configured at the moment if the user includes an image the role does not change and is not redirected anywhere.  The if statement which wraps the image build is separate from the reset of the control flow.

What will work is something like
```
def create
  @listing = current_user.listings.new(listing_params)
  if params["listing_images"]
    params["listing_images"]["images"].each do |i|
      @listing.listing_images.build(image: i)
    end
  end
  if @listing.save
    current_user.update_attributes(role: 1)
    redirect_to user_listing_path(current_user.slug, @listing.id)
  else
    render :new
  end
end
```